packages/compiler/core/checker.ts

```diff
@@ -2265,16 +2264,32 @@ export function createChecker(program: Program): Checker {
       return;
     }
 
-    if (inheritedPropertyNames?.has(newProp.name)) {
-      program.reportDiagnostic(
-        createDiagnostic({
-          code: "override-property",
-          format: { propName: newProp.name },
-          target: diagnosticTarget ?? newProp,
-        })
-      );
-
-      return;
+    if (inheritedProperties?.has(newProp.name)) {
+      for (const inheritedProp of inheritedProperties.get(newProp.name)!) {
```

Comment:
nit:
instead of `inheritedProperties?.has(newProp.name)` then `inheritedProperties.get(newProp.name)!` you can just assign the result of `.get` to a variable and check the if there. That make it so you don't have to force saying `.get` is not `undefined`